[JP] Basic kanji and some other stuff

bluetoad07 opened the pull request:
Back at it again! Very excited to finally start on this :]

This PR will include once finished:
- [x] using references for dakuten/handakuten kana in FontForge
- [ ] more kana redesigns because i can't leave anything alone
- [ ] the following 80 kanji, which are the ones considered necessary for the N5 Japanese Language Learning Proficiency test, the metric I have decided to use for which kanji to include when. (A few of these are already in the font from my last PR):
``日一国人年大十二本中長出三時行見月分後前生五間上東四今金九入学高円子外八六下来気小七山話女北午百書先名川千水半男西電校語土木聞食車何南万毎白天母火右読友左休父雨``
- [ ] probably some other stuff

**References:** Kana characters with dakuten/handakuten (``゛゜``) now make use of FontForge's reference feature. This has no effect on the look of the font but will make it much easier to make edits to them, as well as keeping any kanji based on them consistent in the future.

This PR is a work in progress :3

PhMajerus commented:
Remember I have Naga10 (9×9) and Shinonome12 (~11×11), both containing the complete JIS X 0208 set of 6355 kanjis converted to .bdf and adjusted to fit Cozette.
I still want to also get M⁺ Bitmap Fonts by Coji Morishita (森下 浩司), another 9×9 font, working as a .bdf adjusted for Cozette to be able to check out which one works best.

If you want to experiment with them to check out kanjis sizes, you can import each into Cozette.sfd and check them out :
[Cozette kanjis from other fonts by PHM.zip](https://github.com/user-attachments/files/20130488/Cozette.kanjis.from.other.fonts.by.PHM.zip)

Remember I didn't check all the licensing issues yet, this would need to be validated before including either of those in Cozette.

bluetoad07 commented:
Pretty sure the kana updates are done! Most of these have reworked designs, here is the new character map:

![kana_updatess](https://github.com/user-attachments/assets/5bf8ea66-6030-4c50-a7ad-73283f0c3979)

This is the first time I'm 100% happy with the look of the character set as a whole. No more out-of-place looking characters that i couldn't think of anything better for, everything here is done to the best of my ability!

Halfwidth katakana are updated as well to match:

![halfwidth_updates](https://github.com/user-attachments/assets/3711895c-a45a-4707-a35d-de167089a4d1)

I also streamlined the square katakana characters from the CJK Compatibility block added in @PhMajerus's last commit, so that the characters are bigger (consistently 5x5) and evenly spaced:

![compatability_updates](https://github.com/user-attachments/assets/5f6a1ca0-13a8-4996-80dd-108b3d46a68d)
<sup>I'm pretty sure these aren't in common use at all, but I still want them to look as nice as possible. I also might add the rest of them eventually.</sup>

Iteration marks also have updated designs, there is adjusted spacing for lone dakuten and handakuten, and there is the new addition of combining dakuten and handakuten characters.

And with that, I'll probably start work on the kanji next weekend! Thanks <3

bluetoad07 commented:
Update: I think I'm doing 11x11 kanji now. The way they look has grown on me a lot and they will be so much easier to design.

bluetoad07 commented:
@PhMajerus I totally get why I should use the Shinonome12 kanji, but I really want to design them myself. I want these to be both functional and nice looking, and the Shinonome12 kanji aren't my favorite design-wise. Some of them also extend downwards beyond the 11x11 size I want and I don't like how that looks. I found [this search tool](https://fonts.google.com/?subset=japanese&preview.text=%E6%97%A5%E6%9C%AC%E8%AA%9E&preview.text_type=custom) on google fonts that lets me quickly compare how characters look across different fonts which helps a ton with consistency and simplifying strokes when necessary. I find it very fun to design these by hand, so while I get why just using the Shinonome12 kanji would be great I still want to continue with what I'm doing.

As for the .sfd thing, how would I go about doing this? I assume it would be similar to how CozetteCrossedSeven.sfd works, but I don't know how to set up the builder to automate that. If you could link some resources or maybe set that up for me in a PR of your own once this one is done that would be great!

Thanks for your help! This set is almost done, I should have it ready later today.

bluetoad07 commented:
Here they are :D

Cozette now has 88 kanji!!

![yeahh-kanjii](https://github.com/user-attachments/assets/9139ac95-2d9c-4ce1-b295-2b96923fe967)

They are all 11x11, including the few I added in my last PR that were originally 9x9.

For comparison, here's how Shinonome12 kanji look (these are frequently 11x12):

![sad-kanji](https://github.com/user-attachments/assets/7fd568f4-3bb1-4069-801f-61a7d98ff325)

Very happy with how they turned out! I'm gonna wait a few days to mark this ready for review in case there's anything else that it needs.

PhMajerus commented:
If ![image](https://github.com/user-attachments/assets/7bf05aab-c723-4735-ae68-ee13ff750bf2) are 書 (`U+66F8`) and 電 (`U+96FB`), you switched horizontal lines to vertical. I'm not sure how you want to handle them, but they should be more like ![image](https://github.com/user-attachments/assets/ca7f58fb-a65b-485a-8208-8d8a6073956a).

Also, you're mixing rectangles with plain corners:
![image](https://github.com/user-attachments/assets/cea4c308-3797-476c-a924-93f575fb7242)
and some with descending lines:
![image](https://github.com/user-attachments/assets/c7cf7fd1-264d-47ac-97d2-e4628c3fdc82)
I know this is very common, but I'd consider making them all without descenders/sherifs, it seems to fit better with Cozette general style and it would make several of them easier to design as well, for example 日 would become perfectly even.

bluetoad07 commented:
Thanks, these are great improvements! I removed most of the overhanging strokes, but kept them for 右, 名, 四, and 西 to keep them consistant with ロ (I can't remove them from ロ because any other proportions for the square looks very odd.) I also fixed 書 and 電.

![fixed-kanji](https://github.com/user-attachments/assets/a05a88a6-ca3b-4342-9eb9-77aff0bc2bb2)

bluetoad07 commented:
Alright, time to open this for review! here's everything changed:

Characters added: ``一七三上下九二五今仝休何先入八六出前北十午半南友右名四国外天女子字学山川左後書本来東校母毎気漢父男白聞行西見話語読車長間雨電食高``

Characters updated: ``うぇえかがきぎけげこごさざしじすずせぜそぞただちぢっつづてでとどなにのはばぱひびぴふほぼぽみむもゃやゅゆょよらりるろんゕゖアォオカガクグサザシジスズソゾタダチヂッツヅテデトドナネノハバパマメモャヤュユリルレロン㌃㌍㌔㌘㌢㌣㌦㌧㌫㌶㌻㍉㍊㍍㍑万中人円分区千土大小市年日時月木村水火生町百秒金㊀㊁㊂㊃㊄㊅㊆㊇㊈㊉㋿㍻㍼㍽㍾``

Thank you!!

bluetoad07 commented:
This is taking a while to be merged so I made some small changes :3

![kanji-revisionss](https://github.com/user-attachments/assets/ee4b4c59-5a83-4fb7-9eba-7e1f4604c5d2)

I've been studying stroke order a lot recently and it made me rethink a few designs. I also updated the NerdFonts Arch logo to look nicer just because.

@the-moonwitch this is actually done now i swear lol

the-moonwitch commented:
Wonderful work, thank you so much!
